datalad/distribution/subdatasets.py

```diff
@@ -56,67 +66,68 @@
 valid_key = re.compile(r'^[A-Za-z][-A-Za-z0-9]*$')
 
 
-def _parse_gitmodules(dspath):
-    gitmodule_path = opj(dspath, ".gitmodules")
-    parser = GitConfigParser(gitmodule_path)
+def _parse_gitmodules(ds):
+    gitmodules = ds.pathobj / '.gitmodules'
+    if not gitmodules.exists():
+        return {}
+    # pull out file content
+    out, err = ds.repo._git_custom_command(
+        '',
+        ['git', 'config', '-z', '-l', '--file', '.gitmodules'])
+    # abuse our config parser
+    db, _ = _parse_gitconfig_dump(out, {}, None, True)
     mods = {}
-    for sec in parser.sections():
-        try:
-            modpath = parser.get(sec, 'path')
-        except Exception:
-            lgr.debug("Failed to get '%s.path', skipping section", sec)
+    for k, v in iteritems(db):
+        if not k.startswith('submodule.'):
+            # we don't know what this is
             continue
-        if not modpath or not sec.startswith('submodule '):
+        k_l = k.split('.')
```

Comment:
suggestion actually to replace 4 lines (can't be done in web ui AFAIK)
```suggestion
        # module name could contain '.' - need to split from the left to get "submodule" (not used)
        # and then from the right to get field
        mod_name, field = k.split('.', 1)[1].rsplit('.', 1)
        mod = mods.get(mod_name, {})
        mod[mod_name] = field
```
I find it difficult to read all the `[X]`s.

<details>
<summary>git is ok with '.' in submodule names so middle parts I guess constitute the mod_name, not just [1] </summary>

```shell
$> datalad create -d . my.subds
[INFO   ] Creating a new annex repo at /tmp/testds/my.subds 
create(ok): my.subds (dataset)                                                                                         
action summary:
  add (ok: 2)
  create (ok: 1)
  save (ok: 1)
(dev) 1 13959 [1].....................................:Fri 17 May 2019 04:31:28 PM EDT:.
(git-annex)hopa:/tmp/testds[master]git-annex
$> cat .gitmodules 
[submodule "my.subds"]
	path = my.subds
	url = ./my.subds
	datalad-id = bf34882e-78e2-11e9-a650-8019340ce7f2
(dev) 1 13960 [1].....................................:Fri 17 May 2019 04:31:32 PM EDT:.
(git-annex)hopa:/tmp/testds[master]git-annex
$> 'git' 'config' '-z' '-l' '--file' '.gitmodules' 
submodule.my.subds.path
(dev) 1 13961 [1].....................................:Fri 17 May 2019 04:31:54 PM EDT:.
(git-annex)hopa:/tmp/testds[master]git-annex
$> git submodule
 6e0a46306893a150df47f7de130fbe74931723ca my.subds (heads/master)
```
</details>

Reply:
Done in a different way.